### PR TITLE
Update embedded property replacement for new jdbc config pattern

### PR DESCRIPTION
**Pull request by labkey-tchad**
#### Rationale
The structure of data source definitions in `application.properties` is changing. We should have a more generic replacement solution to configure a second JDBC data source defined with the new pattern.

Old:
```
#context.dataSourceName[1]=jdbc/@@extraJdbcDataSource@@
#context.driverClassName[1]=@@extraJdbcDriverClassName@@
#context.url[1]=@@extraJdbcUrl@@
#context.username[1]=@@extraJdbcUsername@@
#context.password[1]=@@extraJdbcPassword@@
```
New:
```
#context.resources.jdbc.@@extraJdbcDataSource@@.driverClassName=@@extraJdbcDriverClassName@@
#context.resources.jdbc.@@extraJdbcDataSource@@.url=@@extraJdbcUrl@@
#context.resources.jdbc.@@extraJdbcDataSource@@.username=@@extraJdbcUsername@@
#context.resources.jdbc.@@extraJdbcDataSource@@.password=@@extraJdbcPassword@@
```

#### Related Pull Requests
- https://github.com/LabKey/server/pull/713

#### Changes
- Update embedded property replacement for new jdbc config pattern

### CODE REVIEW

**Comment by labkey-susanh** [comment]
I'm not seeing this version in use in the referenced PR. Should it be?

**Comment by labkey-tchad** [comment]
> I'm not seeing this version in use in the referenced PR. Should it be?

I was going to open a separate PR after Josh merges those changes. This isn't urgent and I might have some other changes for embedded Tomcat deployments.

**Review by labkey-jeckels — COMMENTED** [review]
The changes here look fine to me but I'll defer to the two of you for when and how to merge.

**Review by labkey-susanh — APPROVED** [review]

**Pull request merged**
